homeworks/src/homework13/ValueCalculator.java

```diff
@@ -0,0 +1,71 @@
+package homework13;
+
+public class ValueCalculator {
+
+    private int WORKING_ARRAY_SIZE = 1000000;
+    private int HALF_OF_WORKING_ARRAY_SIZE = WORKING_ARRAY_SIZE / 2;
+    private double[] workingArray = new double[WORKING_ARRAY_SIZE];
+    private double[] workingArrayHalf1 = new double[HALF_OF_WORKING_ARRAY_SIZE];
+    private double[] workingArrayHalf2 = new double[HALF_OF_WORKING_ARRAY_SIZE];
+
+
+    public void comparePerformanceTimes() throws InterruptedException {
+
+        long start = System.currentTimeMillis();
+        long stop;
+
+        System.out.println("Starting to count single thread performance");
+        setWorkingArrays();
+        processWorkingArraysBySingleThread();
+        stop = System.currentTimeMillis();
+        System.out.println("Single thread performance equals " + ((stop - start) / 1000) + " seconds");
+
+        System.out.println("Starting to count multi thread performance");
+        start = System.currentTimeMillis();
+        setWorkingArrays();
+        processWorkingArraysByMultiThread();
+        stop = System.currentTimeMillis();
+        System.out.println("Multi thread performance equals " + ((stop - start) / 1000) + " seconds");
+
+    }
+
+    private void setWorkingArrays() {
+        for (int i = 0; i < WORKING_ARRAY_SIZE; i++) {
+            workingArray[i] = Math.random() * 360;
+            System.arraycopy(workingArray, 0, workingArrayHalf1, 0, HALF_OF_WORKING_ARRAY_SIZE);
+            System.arraycopy(workingArray, HALF_OF_WORKING_ARRAY_SIZE, workingArrayHalf2, 0, HALF_OF_WORKING_ARRAY_SIZE);
+        }
+    }
+
+    private void processWorkingArraysBySingleThread() {
+        for (int i = 0; i < HALF_OF_WORKING_ARRAY_SIZE; i++) {
+            workingArrayHalf1[i] = (double) (workingArrayHalf1[i] * Math.sin(0.2f + i / 5) * Math.cos(0.2f + i / 5) * Math.cos(0.4f + i / 2));
+            workingArrayHalf2[i] = (double) (workingArrayHalf2[i] * Math.sin(0.2f + i / 5) * Math.cos(0.2f + i / 5) * Math.cos(0.4f + i / 2));
```

Comment:
You repeat math operations three times. What will happen, if tomorrow's logic (something in the formula) will be changed?